packages/langium/src/lsp/langium-lsp-module.ts

```diff
@@ -0,0 +1,115 @@
+/******************************************************************************
+ * Copyright 2023 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+******************************************************************************/
+
+import { DefaultCompletionProvider } from './completion/completion-provider.js';
+import { DefaultDocumentHighlightProvider } from './document-highlight-provider.js';
+import { DefaultDocumentSymbolProvider } from './document-symbol-provider.js';
+import { DefaultFoldingRangeProvider } from './folding-range-provider.js';
+import { DefaultDefinitionProvider } from './definition-provider.js';
+import { MultilineCommentHoverProvider } from './hover-provider.js';
+import { DefaultReferencesProvider } from './references-provider.js';
+import { DefaultRenameProvider } from './rename-provider.js';
+import type { ServiceRegistryI, LangiumServices, LangiumLSPServices, LangiumSharedLSPServices, PartialLangiumSharedLSPServices, LangiumSharedServices } from './lsp-services.js';
+import { inject, type Module } from '../dependency-injection.js';
+import { TextDocuments } from 'vscode-languageserver';
+import { createDefaultModule, createDefaultSharedModule, type DefaultSharedModuleContext } from '../default-module.js';
+import { DefaultLanguageServer } from './language-server.js';
+import { DefaultWorkspaceSymbolProvider } from './workspace-symbol-provider.js';
+import { DefaultNodeKindProvider } from './node-kind-provider.js';
+import { DefaultFuzzyMatcher } from './fuzzy-matcher.js';
+import { TextDocument } from 'vscode-languageserver-textdocument';
+import type { LangiumSharedCoreServices, PartialLangiumSharedCoreServices } from '../services.js';
+import { LangiumGrammarModule, type LangiumGrammarServices } from '../grammar/langium-grammar-module.js';
+import { LangiumGrammarGeneratedModule, LangiumGrammarGeneratedSharedModule } from '../grammar/generated/module.js';
+import { registerValidationChecks } from '../grammar/internal-grammar-util.js';
+import { registerTypeValidationChecks } from '../grammar/validation/types-validator.js';
+import { LSPConfigurationProvider } from './workspace/default-configuration.js';
+import { LSPLangiumDocumentFactory } from './workspace/documents.js';
+import { LSPWorkspaceManager } from './workspace/workspace-manager.js';
+import { CoreServiceRegistry } from '../service-registry.js';
+
+/**
+ * Context required for creating the default language-specific dependency injection module.
+ */
+export interface LSPModuleContext {
+    shared: LangiumSharedLSPServices;
+}
+
+/**
+ * Create a dependency injection module for the default LSP services. This is a
+ * set of services that are used by exactly one language.
+ */
+export function createLSPModule(context: LSPModuleContext): Module<LangiumServices, LangiumLSPServices> {
```

Comment:
similarly I would prefer the name `createDefaultLSPModule` (see comment on line 1)

Reply:
See above

Reply:
Also applied this in an upcoming commit.